### PR TITLE
[BPF] set MTU on bpfin/out.cali to match host

### PR DESCRIPTION
**Pull request by tomastigera**
Set the MTU to match the smallest MTU of the host devices as per the current autodetection. That makes sure that if large MTU is used, the extra device does not create a bottleneck with a small MTU. Also when MTU changes, on host devices, it get adjusted automatically.  If an overlay is used, the special device MTU is adjusted to the size of the overlay.

Fixes https://github.com/projectcalico/calico/issues/8868

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by fasaxc — CHANGES_REQUESTED** [review]
Could do with a test, but LGTM otherwise

**Comment by tomastigera** [comment]
> Could do with a test, but LGTM otherwise

enabled VXLAN mtu tests that now also check the bpf special device mtu which is set to the overlay mtu

**Comment by tomastigera** [comment]
> Could do with a test, but LGTM otherwise

I enabled the vxlan mtu test for vxlan and it now also tests the bpfin.cali mtu

File: felix/dataplane/linux/bpf_ep_mgr.go
Comment (fasaxc) [inline felix/dataplane/linux/bpf_ep_mgr.go:461]: Called podMTU one later up, maybe bring them into sync or make this `bpfIfaceMTU` so it's more clear what it's for?

**Review by fasaxc — APPROVED** [review]
One nit but LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by marvin-tigera** [comment]
Removing "merge-when-ready" label due to new commits

**Comment by tomastigera** [comment]
The only error has a fix here https://github.com/projectcalico/calico/pull/8952

**Pull request merged**
